lib/class-wp-theme-json-gutenberg.php

```diff
@@ -4085,4 +4085,35 @@ protected static function get_valid_block_style_variations() {
 
 		return $valid_variations;
 	}
+
+	/**
+	 * Resolves relative paths in theme.json styles to theme absolute paths.
+	 *
+	 * @since 6.6.0
+	 *
+	 * @return WP_Theme_JSON_Gutenberg The current instance of $this.
+	 */
+	public function resolve_theme_file_uris() {
+		if ( empty( $this->theme_json ) ) {
+			return $this;
+		}
+
+		/*
+		 * Styles backgrounds.
+		 * Where a URL is not absolute (has no host fragment), it is assumed to be relative to the theme directory.
+		 * Blocks, elements, and block variations are not yet supported.
+		 */
+		if (
+			isset( $this->theme_json['styles']['background']['backgroundImage']['url'] ) &&
+			is_string( $this->theme_json['styles']['background']['backgroundImage']['url'] ) &&
+			! isset( wp_parse_url( $this->theme_json['styles']['background']['backgroundImage']['url'] )['host'] ) ) {
+				_wp_array_set(
+					$this->theme_json,
+					array( 'styles', 'background', 'backgroundImage', 'url' ),
+					esc_url( get_theme_file_uri( $this->theme_json['styles']['background']['backgroundImage']['url'] ) )
+				);
+		}
```

Comment:
Just wondering, if we're adding this logic to the theme JSON class, if and when we go to support referencing theme assets (e.g. background images) at the block level within patterns, will we need to duplicate this logic in `background.php`?

I guess to put the question another way — is it worth abstracting into a function in the background image support at this stage, or leave it inline if we know we're going to be using the `resolve_theme_file_uris()` function anyway, and the innards can be moved elsewhere further down the track?

From my perspective, looking at the scope of this PR, if we know we're happy with the `resolve_theme_file_uris()` function and signature, then it sounds reasonable to move things around in follow-ups to me.

Reply:
> if and when we go to support referencing theme assets (e.g. background images) at the block level within patterns, will we need to duplicate this logic in background.php (or have similar logic over there)?

Yeah we'll have to do that, so abstracting should be 👍🏻 

What do you think? We'd move the logic into a `background.php` function and call it from theme.json? We do it with typography and duotone already so 🤷🏻 For global styles, ideally the transformation would occur in `WP_Theme_JSON_Resolver::get_merged_data` as it is in this PR, so it's "resolved" everywhere. 

I can reinstate this function (again) 😄 

https://github.com/WordPress/gutenberg/blob/eca8d3cacc96efb30a6da73e05aef0bf302b0c74/lib/block-supports/background.php#L41C10-L41C49

Reply:
> We'd move the logic into a background.php function and call it from theme.json?

Sorry, typing too fast. I wanted to also write, "or, call the block supports function directly in `P_Theme_JSON_Resolver::get_merged_data`

Reply:
> What do you think? We'd move the logic into a background.php function and call it from theme.json? 

Yeah, the example in your comment above looks pretty good to me — keeps the theme.json class quite light, and parallels the approach the typography support uses as you mention. IMO I like the structure where the block supports have functions for their own logic and the theme json calls out to them when needed, but that could just be my own preference!